packages/server/email/ServerEnvironment.ts

```diff
@@ -41,7 +42,8 @@ export default class ServerEnvironment extends Environment {
           dataLoaderId: this.dataLoaderId
         })
       )
-      return undefined as any
+      // relay expects an array of responses, or a single valid response
+      return [] as any
```

Comment:
arrays are accepted for multi-part responses, so relay hooks handle empty arrays correctly, but cannot handle `undefined`, which makes sense, because the server would never send it that response.